Import suggestion breaks `#[cfg]` attribute for previous imports

### Code

```Rust
#[cfg(unix)]
use std::os::unix::fs;

fn main() {
    let _ = stdout();

    #[cfg(unix)]
    {
        // Using unix specific stuff here.
        let _function = fs::chroot::<&str>;
    }
}
```


### Current output

```Shell
error[E0425]: cannot find function `stdout` in this scope
 --> src/main.rs:5:13
  |
5 |     let _ = stdout();
  |             ^^^^^^ not found in this scope
  |
help: consider importing this function
  |
2 + use std::io::stdout;
  |
```


### Desired output

```Shell
...
help: consider importing this function
  |
3 + use std::io::stdout;
  |

((( with `3` instead of `2` )))
```


### Rationale and extra context

After applying the suggestion, you end up with this code:

```rust
#[cfg(unix)]
use std::io::stdout;
use std::os::unix::fs;

fn main() {
    let _ = stdout();

    #[cfg(unix)]
    {
        // Using unix specific stuff here.
        let _function = fs::chroot::<&str>;
    }
}
```

Note that the `cfg(unix)` attribute is at the wrong place, so code won't compile on Windows anymore.

I'd say the ideal solution is to either:
- Suggest import after the last import line.
- Suggest import following some other rule where it's guaranteed to not fall under an attribute.

### Other cases

_No response_

### Anything else?

Comes from https://github.com/rust-lang/rust-enhanced/issues/516 (Sublime Text's "extension" for Rust).

The extension relies on `rustc`'s import suggestions to apply fixes (user is prompted), accepting the fix breaks your code.

Useful label: `D-invalid-suggestion`

--- a/compiler/rustc_resolve/src/diagnostics.rs
+++ b/compiler/rustc_resolve/src/diagnostics.rs
@@ -2753,7 +2753,13 @@ fn search_for_any_use_in_items(items: &[P<ast::Item>]) -> Option<Span> {
     for item in items {
         if let ItemKind::Use(..) = item.kind {
             if is_span_suitable_for_use_injection(item.span) {
-                return Some(item.span.shrink_to_lo());
+                let mut lo = item.span.lo();
+                for attr in &item.attrs {
+                    if attr.span.eq_ctxt(item.span) {
+                        lo = std::cmp::min(lo, attr.span.lo());
+                    }
+                }
+                return Some(Span::new(lo, lo, item.span.ctxt(), item.span.parent()));
             }
         }
     }

--- a/tests/ui/resolve/suggest-import-without-clobbering-attrs.fixed
+++ b/tests/ui/resolve/suggest-import-without-clobbering-attrs.fixed
@@ -1,0 +1,16 @@
+// run-rustfix
+// compile-flags: --cfg=whatever -Aunused
+
+#[cfg(whatever)]
+use y::z;
+use y::Whatever;
+
+mod y {
+    pub(crate) fn z() {}
+    pub(crate) struct Whatever;
+}
+
+fn main() {
+    z();
+    //~^ ERROR cannot find function `z` in this scope
+}

--- a/tests/ui/resolve/suggest-import-without-clobbering-attrs.fixed
+++ b/tests/ui/resolve/suggest-import-without-clobbering-attrs.fixed
@@ -1,8 +1,8 @@
 // run-rustfix
 // compile-flags: --cfg=whatever -Aunused
 
-#[cfg(whatever)]
 use y::z;
+#[cfg(whatever)]
 use y::Whatever;
 
 mod y {

--- a/tests/ui/resolve/suggest-import-without-clobbering-attrs.rs
+++ b/tests/ui/resolve/suggest-import-without-clobbering-attrs.rs
@@ -1,0 +1,15 @@
+// run-rustfix
+// compile-flags: --cfg=whatever -Aunused
+
+#[cfg(whatever)]
+use y::Whatever;
+
+mod y {
+    pub(crate) fn z() {}
+    pub(crate) struct Whatever;
+}
+
+fn main() {
+    z();
+    //~^ ERROR cannot find function `z` in this scope
+}

--- a/tests/ui/resolve/suggest-import-without-clobbering-attrs.stderr
+++ b/tests/ui/resolve/suggest-import-without-clobbering-attrs.stderr
@@ -1,0 +1,14 @@
+error[E0425]: cannot find function `z` in this scope
+  --> $DIR/suggest-import-without-clobbering-attrs.rs:13:5
+   |
+LL |     z();
+   |     ^ not found in this scope
+   |
+help: consider importing this function
+   |
+LL + use y::z;
+   |
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0425`.

